### PR TITLE
Rename Inception V2 to BN-Inception.

### PR DESCRIPTION
**Pull request by dichotomies**
As discussed in https://github.com/tensorflow/models/issues/5054, I'm opening a initial PR. I've only edited the readme file, but further changes will be necessary.

The problem I'm addressing is that the link of Inception V2 in the readme file refers [Batch Normalization: Accelerating Deep Network Training by Reducing Internal Covariate Shift](https://arxiv.org/abs/1502.03167). The implemented [network architecture](https://github.com/tensorflow/models/blob/master/research/slim/nets/inception_v2.py) has the same output sizes as in the paper (p. 11), so I think that the reference to the paper is correct. It is named BN-Inception.

Inception V2 (with V3) was introduced in [Rethinking the Inception Architecture for Computer Vision](https://arxiv.org/abs/1512.00567) and it has a different network architecture compared to the implemented one (see p. 6).

Therefore I think that the name of the network architecture in the repository should be changed from Inception V2 to BN-Inception.

### CODE REVIEW

**Comment by dichotomies** [comment]
I signed the CLA.

**Comment by karmel** [comment]
Thanks @dichotomies -- though I think I misunderstood the intent of the original issue. I thought you were suggesting changing/adding a paper reference, not renaming the model altogether. Given the existing links to InceptionV2 as it is named, it will be hard to attempt a rename for the entirety. Maybe adding an additional link and note to the paper is more reasonable? @sguada , can you comment?

**Comment by dichotomies** [comment]
@karmel While submitting the PR, I've noticed that the actual problem is not the link to the paper (links to BN-Inception), but the name of the link, i.e. the implemented model seems to be BN-Inception and not Inception V2. So you haven't misunderstood it, my initial guess was just inaccurate.

**Comment by karmel** [comment]
Given historical references to the name as-is, I suspect it would be more confusing to rename now than to not. In that case, let's leave as-is. If you would like to add a note or comment referring to BN-inception, we can do that.

**Comment by dichotomies** [comment]
Something like: "Note that the correct model name is BN-Inception, but it was defined as Inception V2 when implemented. Due to historical references to the name as-is, it might be more confusing to rename it now. Therefore, the name is left unchanged in the code and is only corrected in this README."?
